docker: init (#497)
gitpod-io/website: src/docs/docker-in-gitpod.md

# Docker in Gitpod

> Theia on gitpod is currently provided in docker containers where this page is dedicated to makeing a dockerimages for gitpod and to work with dockerfiles.

Reviewer: Sorry, I don't understand this sentence.

Alternative suggestion:

```suggestion
Gitpod allows writing, building and debugging Dockerfiles. This page is here to help you work with Dockerfiles in Gitpod, and to further enhance your Docker developer experience.
```

### What are dockerfiles

> FIXME-DOCS

Reviewer: Please fix it.

### Support of dockerfiles in gitpod
See [Docker Image](src/docs/config-docker.md)

> ### Development of docker
> Currently blocked by https://github.com/gitpod-io/gitpod/issues/52

Reviewer: This is not a good section, because by reading it I assume that the development of Docker itself is blocked by some Gitpod issue.

It should be removed, or clarified:

```suggestion
```

> ### Linting of dockefiles

Reviewer: Typo:

```suggestion
### Linting of dockerfiles
```

Linting of dockerfiles can be done using [Hadolint](https://www.google.com/search?client=firefox-b-e&q=hadolint) which can be installed as vscode extension on https://marketplace.visualstudio.com/items?itemName=exiasr.hadolint for gitpod to provide realtime linting.

![](https://i.gyazo.com/a701460ccdda13a1a449b2c3e8da40bc.gif)

Note that for this extension to work you need a hadolint executable which is available in homebrew and as binary on https://github.com/hadolint/hadolint/releases.

The binary can be installed in gitpod's dockerfile the following is example:
```dockerfile
RUN wget https://github.com/hadolint/hadolint/releases/download/v1.17.5/hadolint-Linux-x86_64 -O /usr/bin/hadolint && chmod +x /usr/bin/hadolint
```

As of 24th February of 2020 this executable is not available in downstream of debian and ubuntu.

### DEMO

> Not sure about Bash IDE? Try it in Gitpod

Reviewer: What is Bash IDE and why is it relevant here?

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/JesterOrNot/Gitpod-BashIDE)

## External Resources

> - For more on ShellCheck see https://shellcheck.net

Reviewer: This is the first time you mention ShellCheck in this article, so "Fore more on" doesn't make sense:

```suggestion
- To lint Bash scripts in Dockerfiles, please take a look at [ShellCheck](https://shellcheck.net)
```

(Or remove.)
